lib/src/layer/tile_layer.dart

```diff
@@ -401,55 +640,101 @@ class _TileLayerState extends State<TileLayer> {
     if (_wrapX != null) {
       var first =
           (map.project(LatLng(0.0, crs.wrapLng.item1), tileZoom).x / tileSize.x)
-              .floor()
-              .toDouble();
+              .floorToDouble();
       var second =
           (map.project(LatLng(0.0, crs.wrapLng.item2), tileZoom).x / tileSize.y)
-              .ceil()
-              .toDouble();
+              .ceilToDouble();
       _wrapX = Tuple2(first, second);
     }
 
     _wrapY = crs.wrapLat;
     if (_wrapY != null) {
       var first =
           (map.project(LatLng(crs.wrapLat.item1, 0.0), tileZoom).y / tileSize.x)
-              .floor()
-              .toDouble();
+              .floorToDouble();
       var second =
           (map.project(LatLng(crs.wrapLat.item2, 0.0), tileZoom).y / tileSize.y)
-              .ceil()
-              .toDouble();
+              .ceilToDouble();
       _wrapY = Tuple2(first, second);
     }
   }
 
-  double _clampZoom(double zoom) {
-    // todo
-    return zoom;
+  void _handleMove() {
+    var tileZoom = _clampZoom(map.zoom.roundToDouble());
+
+    if (_tileZoom == null) {
+      // if there is no _tileZoom available it means we are out within zoom level
+      // we will restory fully via _setView call if we are back on trail
+      if ((options.maxZoom != null && tileZoom <= options.maxZoom) &&
+          (options.minZoom != null && tileZoom >= options.minZoom)) {
+        _tileZoom = tileZoom;
+        setState(() {
+          _setView(map.center, tileZoom);
+        });
+      }
+    } else {
+      setState(() {
+        if ((tileZoom - _tileZoom).abs() >= 1) {
+          // It was a zoom lvl change
+          _setView(map.center, tileZoom);
+        } else {
+          if (null == _throttleUpdate) {
```

Comment:
@maRci002 do you always put null on the left hand side of an equality check? :P

Reply:
Sometimes I like using [yoda style](https://en.wikipedia.org/wiki/Yoda_conditions) :smile:

It has it own advantage; forinstance in JS this is a valid expression statement (but user wanted to check if `i` equals to `7`, like `i == 7` or `i === 7`)
```
var i = 0;
if (i=7){
  // this block will be executed
}
console.log(i); // prints 7
```

Or in java:
```
String i = null;
if (i.equals("CONSTANT")) {} // NullPointerException
```

It has it's own benefits but not in Dart :sweat_smile: